fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SubspaceProvider.java

```diff
@@ -34,17 +35,23 @@
 public interface SubspaceProvider {
```

Comment:
Hm, I'm not sure how I feel about breaking compatibility with this interface. We generally want `MAINTAINED` interfaces not to break except at least between minor versions. I suppose we could could go to version 2.6 with this change.

It might also be a sign that this is really an `UNSTABLE` interface. I don't think that's too bad given that the user can usually avoid using it by providing a `KeySpacePath` or a `Subspace` instead.

Reply:
So what do we want to do here? Mark the interface UNSTABLE and go with the breaking changes? Keep it MAINTAINED and break it now? (toString() is going to break this week as well)? Other?

Reply:
I think I could live with it as `UNSTABLE` and accept the breaking change if we don't want to do something like what was done to the `KeySpacePath` API where the old methods were deprecated. (Marking things as deprecated would be my preference given infinite resources, though it will certainly make this API ugly.)

Keeping it `MAINTAINED` with the foreknowledge that we intend to make further breaking changes seems wrong, though the `toString` thing is a little different given that the user can continue calling `toString` as removing a method from the `Object` API is far beyond our poor power to add or detract.